src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -2412,135 +2448,104 @@ public override BoundNode VisitConditionalOperator(BoundConditionalOperator node
-                        alternativeResult);
+                        alternativeRValue);
                 }
 
-                if (!convertedAlternativeResult.HasType)
-                {
-                    Debug.Assert(convertedConsequenceResult.HasType);
-                    visitResult = convertedConsequenceResult;
-                }
-                else if (!convertedConsequenceResult.HasType)
+                resultState = (consequenceReachable, alternativeReachable) switch
                 {
-                    Debug.Assert(convertedAlternativeResult.HasType);
-                    visitResult = convertedAlternativeResult;
-                }
-                else
-                {
-                    visitResult = TypeWithAnnotations.Create(resultType,
-                        convertedConsequenceResult.NullableAnnotation.Join(convertedAlternativeResult.NullableAnnotation));
-                }
-            }
-            else
-            {
-                NullableAnnotation resultNullableAnnotation;
-
-                if (isConstantTrue)
-                {
-                    resultNullableAnnotation = getNullableAnnotation(consequence, consequenceResult);
-                }
-                else if (isConstantFalse)
-                {
-                    resultNullableAnnotation = getNullableAnnotation(alternative, alternativeResult);
-                }
-                else
-                {
-                    resultNullableAnnotation = getNullableAnnotation(consequence, consequenceResult).Join(getNullableAnnotation(alternative, alternativeResult));
-                }
-
-                visitResult = TypeWithAnnotations.Create(node.Type.SetUnknownNullabilityForReferenceTypes(), resultNullableAnnotation);
+                    (false, false) => NullableFlowState.NotNull,
+                    (false, _) => convertedAlternativeResult.State,
+                    (_, false) => convertedConsequenceResult.State,
+                    _ => convertedConsequenceResult.State.Join(convertedAlternativeResult.State)
+                };
             }
-            LvalueResultType = visitResult;
 
+            ResultType = new TypeWithState(resultType, resultState);
             return null;
 
-            NullableAnnotation getNullableAnnotation(BoundExpression expr, TypeWithAnnotations type)
+            (BoundExpression, Conversion, TypeWithState) visitConditionalOperand(LocalState state, BoundExpression operand)
             {
-                if (type.HasType)
-                {
-                    return type.GetValueNullableAnnotation();
-                }
-                if (expr.IsLiteralNullOrDefault())
+                Conversion conversion;
+                SetState(state);
+                Debug.Assert(!node.IsRef);
+
+                (operand, conversion) = RemoveConversion(operand, includeExplicitConversions: false);
+                Visit(operand);
+                if (node.HasErrors)
```

Comment:
`|| !State.IsReachable`

Reply:
I think `Visit(operand)` should already take care of that

---
In reply to: [268371847](https://github.com/dotnet/roslyn/pull/34323#discussion_r268371847) [](ancestors = 268371847)

Reply:
Actually, Visit should do the `HasErrors` check too.

---
In reply to: [268373629](https://github.com/dotnet/roslyn/pull/34323#discussion_r268373629) [](ancestors = 268373629,268371847)